### PR TITLE
Add: Legacy UI test parameter inputs

### PR DESCRIPTION
**Pull request by taqtiqa-mark**
Address issue #81.

This test reflects the current state of play.

Migrating to the pipeline set out in issue #113 will enable this
to be replaced by the more informative text and error location:

```
error: this attribute takes one argument

         = help: use `#[trace("span-name")]`

 --> tests/ui/has-arguments.rs:3:1
  |
3 | #[trace(a, b)]
  | ^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `trace`
    (in Nightly builds, run with -Z macro-backtrace for more info)
```

### CODE REVIEW

File: minitrace-macro/Cargo.toml [hunk 1]
```diff
@@ -20,3 +20,10 @@ syn = { version = "1.0.84", features = ["full", "parsing", "extra-traits", "proc
 quote = "1"
 proc-macro2 = "1"
 proc-macro-error = "1"
+
+[dev-dependencies]
+trybuild = "1"
+minitrace = "0.4.0"
+
+[patch.crates-io]
+minitrace = { path = "../" }
```
Comment (zhongzc) [inline minitrace-macro/Cargo.toml:29]:
```suggestion
minitrace = { path = "../" }

```

File: minitrace-macro/tests/ui.rs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#[test]
+fn ui() {
+    let t = trybuild::TestCases::new();
+    t.compile_fail("tests/ui/err/*.rs");
+    // t.pass("tests/ui/ok/*.rs");
```
Comment (zhongzc) [inline minitrace-macro/tests/ui.rs:5]: Is it a todo?

**Review by zhongzc — COMMENTED** [review]
Rest LGTM

File: minitrace-macro/tests/ui.rs (hunk 2 above)
Comment (taqtiqa-mark) [reply minitrace-macro/tests/ui.rs:5]: Yes, these will be landing shortly.

File: minitrace-macro/Cargo.toml (hunk 1 above)
Comment (taqtiqa-mark) [reply minitrace-macro/Cargo.toml:29]: Thanks for taking the time to review these changes.

I am not sure I follow this suggested change. 

To clarify, you prefer not to use Cargo's `[patch.cargo.io]` interface, and would rather replace:

```toml
[dev-dependency]
...
minitrace = 0.4.0
...
```

with 

```toml
[dev-dependency]
...
minitrace = { version = "0.4.0", path = "../"}
...
```

Correct?

Comment (taqtiqa-mark) [reply minitrace-macro/Cargo.toml:30]: OK, I see the patch is being ignored:

https://github.com/tikv/minitrace-rust/runs/5625725597?check_suite_focus=true#step:4:11

**Comment by taqtiqa-mark** [comment]
@andylokandy as best I can tell this lint failure:

https://github.com/tikv/minitrace-rust/runs/5625725597?check_suite_focus=true#step:4:430

Is caused by these GitHub workflow settings:

https://github.com/tikv/minitrace-rust/blob/ede1ec506c2fc9f391f6c37489ab1cccdae5c2bb/.github/workflows/coverage.yml#L21-L22

Since I can't trigger these runs I'm going have to leave this in the lap of yourself or whoever...

**Review by zhongzc — APPROVED** [review]

**Pull request merged**